codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -39,22 +39,30 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.HashMap;
 import java.util.HashSet;
-import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
 
 /**
  * Simple implementation of {@link Http2Connection}.
  */
 public class DefaultHttp2Connection implements Http2Connection {
 
+    private static final Comparator<Http2Stream> STREAMID_COMPARATOR = new Comparator<Http2Stream>() {
+        @Override
+        public int compare(Http2Stream o1, Http2Stream o2) {
+            return o1.id() - o2.id();
```

Comment:
> If I understood the priority related code in the flow controller correctly, it computes the each stream's share of the available connection window

Yes.

> So I then thought it would make sense to also flush the "higher priority" streams first, but that's okay I guess

This is a good idea and one I tried to implement while working on the tree algorithms but it was much more expensive than just a final pass over the `activeStreams`).  I was maintaining a collection throughout the traversal to guarantee that writes were done in the order they were allocated in the traversal.  Because there is no clear benefit of doing this, and the speed/memory usage improved by just using `activeStreams` I just resorted to using this to ensure all allocations were eventually written.  Maintaining a "total priority order" outside of this tree traversal would be complicated and likely expensive, and so I don't think we want to do this.